### ISSUE TITLE
Set ECM using option on a flight by flight basis.

### ISSUE THREAD

**Issue by SnappyComebacks** [dcs-liberation/dcs_liberation#1878]
Apparently for AI to use ECM the option must be set in the ME.
It would be a good idea to have this be set for the flight types that make sense.

**Is your feature request related to a problem? Please describe.**
AI don't use ECM without using the `ECM Using` option.  (unverified).  Source:  https://www.youtube.com/watch?v=AItrvNEqQp0

**Describe the solution you'd like**
Have ECM be enabled for the flights that make sense.  Picking the best option where appropriate.
However, I don't know what happens if the AI is set to use ECM without the ability to actually use ECM.

**Additional context**
![image](https://user-images.githubusercontent.com/74509817/147607331-270450d6-7c5b-4aab-964e-7f97ceda9f97.png)

**Comment by DanAlbert** [comment]
When is it ever not appropriate? This seems like something that has a default per mission type and there's no reason to ever change it.

**Comment by SnappyComebacks** [comment]
> default per mission type

This is more or less what I was initially thinking.  It doesn't seem to make sense to have it blanket set for all waypoints though, because there is a lot of different radars in Liberation campaigns.  If `USE IF DETECTED OR LOCK BY RADAR` is selected, the AI will always be jamming, because there is always a radar to detect it.  For the case of SEAD, it probably makes sense to have the `ingress` and `target` waypoint be set to `USE IF DETECTED OR LOCK BY RADAR`, but every other waypoint just be `USE ONLY IF LOCKED`.  Otherwise the AI is just highlighting themselves, as its easier to find a jamming target than a non jamming one.

It may be inappropriate to enable jamming on SAMs that have `home on jam` capability though.  However its a tradeoff, as jamming lets the flight get closer.

I set up a scenario to see what the AI does.  Player in a F-15C and hostile with the CAP task in an F-16C with ECM pod at 100nm seperation, 20k feet altitude.

- `ALWAYS USE` does seem to mean that the jammer is always on.
- Just have to trust that `USE IF DETECTED OR LOCK BY RADAR` is what it says I guess (ECM always appeared to be on to me).
- `USE ONLY IF LOCKED` will only jam if you use a `STT` lock.  No jamming if using `TWS` to lock.
- `NEVER USE` does what it says.

If I take away the hostile's ECM pod then the AI never jams.  (So there is no need to consider aircraft loadout which is nice.)

**Comment by DanAlbert** [comment]
> This is more or less what I was initially thinking. It doesn't seem to make sense to have it blanket set for all waypoints though

Yeah, SGTM. Seems like something we can handle completely automatically rather than it being a player-involved thing (if there's anything a player wants to change it most likely means our default behavior is wrong, and rather than adding UI we should just fix the behavior so red can benefit too). I'd originally read this as asking for it to be a UI option, but I think that's just me misunderstanding :)

> It may be inappropriate to enable jamming on SAMs that have home on jam capability though

Do any SAMs in the game do that? If so and we want to avoid that, we know where the SAMs are and what their ranges are, so we can always check the flight path leg against what SAMs are in the area.

**Comment by SnappyComebacks** [comment]
> Do any SAMs in the game do that? If so and we want to avoid that, we know where the SAMs are and what their ranges are

I do not know that one.  This would be tough to do, as you would need to consider if the flight has the ability to jam, and the average burn through range of the jammer.  (Or you can assume that burn through range = range of sam).

**Comment by Starfire13** [comment]
I would think that the SAM would (hopefully) be dead before they burn through the jammer. And if not, that flight was probably never gonna take that SAM out anyway.

**Comment by SnappyComebacks** [comment]
> 



> I would think that the SAM would (hopefully) be dead before they burn through the jammer. And if not, that flight was probably never gonna take that SAM out anyway.

I believe Dan was referring to flight paths that would take you near, or through SAM coverage zones, where the SAM was not the primary target.

**Comment by DanAlbert** [comment]
Yes, although I think any of that is probably overkill at the start :) A patch to automatically set either of the middle two options would go a long way, tuning based on flight type would improve a fair amount, tuning based on waypoint type might be a small improvement, and tuning based on regional SAMs is the possibly unnecessary stretch goal.

**Comment by SnappyComebacks** [comment]
Potential problem.  It seems you can bully an F/A-18C into not firing AIM-120s by STTing it with any of the jamming enabled options (it will fire AIM-9s if it gets close enough).  I suspect this is due to the radar also sharing the role of the jammer.  A F-16C will fire it's AIM-120s, and a SU-27 will turn off its jammer when in range and then fire R-27ERs.  Is this sort of thing a blocker?  Should it be reported as a DCS bug?

**Comment by DanAlbert** [comment]
Yeah, that's how the hornet SPJ works. If it's on the radar doesn't function. I suppose that does mean we need to do this as flight-specific up front. It'd be nice if DCS had a bit more fidelity here so that an A2G flight would turn off its jammer if forced into A2A combat, but I think the way we have the AI configured they go defensive rather than fighting back anyway.
